File: backend/registrations/getRegistrations.mjs
import { QueryCommand } from "@aws-sdk/lib-dynamodb";
import { db, REGISTRATIONS_TABLE } from "./shared/db.mjs";
import { ok, badRequest, serverError, cors } from "./shared/response.mjs";

export async function handler(event) {
  if (event.httpMethod === "OPTIONS") return cors();

  const { email } = event.pathParameters ?? {};
  if (!email) return badRequest("email is required");

  const decoded = decodeURIComponent(email).toLowerCase().trim();

  try {
    const result = await db.send(new QueryCommand({
      TableName: REGISTRATIONS_TABLE,
      IndexName: "email-eventId-index",
      KeyConditionExpression: "email = :email",
      ExpressionAttributeValues: { ":email": decoded },
    }));
    return ok(result.Items ?? []);
  } catch (e) {
    console.error(e);
    return serverError();
  }
}

File: backend/registrations/getRegistrations.py
import logging
from urllib.parse import unquote

from shared.db import registrations_table
from shared.response import bad_request, cors, ok, server_error

log = logging.getLogger()


def handler(event, context):
    if event.get("httpMethod") == "OPTIONS":
        return cors()

    email = (event.get("pathParameters") or {}).get("email")
    if not email:
        return bad_request("email is required")

    decoded = unquote(email).strip().lower()

    try:
        result = registrations_table().query(
            IndexName="email-eventId-index",
            KeyConditionExpression="email = :email",
            ExpressionAttributeValues={":email": decoded},
        )
        return ok(result.get("Items", []))
    except Exception:
        log.exception("getRegistrations failed")
        return server_error()
